app/models/concerns/orchestration/ssh_provision.rb

```diff
@@ -32,10 +32,9 @@ def queue_ssh_provision_update
 
   def setSSHProvisionScript
     logger.info "About to start post launch script on #{name}"
-    template   = provisioning_template(:kind => "finish")
-    @host      = self
+    template = provisioning_template(:kind => "finish")
     logger.info "generating template to upload to #{name}"
-    self.template_file = unattended_render_to_temp_file(template)
+    self.template_file = Foreman::Renderer.render_template_to_tempfile(template: template, prefix: id.to_s)
```

Comment:
this no longer sets @host locally but does not pass the object to render_template_to_file, therefore it does not set @host and breaks template rendering